### PR TITLE
Bail out if being installed in a submodule

### PR DESCRIPTION
**Pull request by all3fox**
Refs #79

### CODE REVIEW

File: install.js [hunk 1]
```diff
@@ -18,7 +18,10 @@ function xmkdirSync (path, mode) {
       console.error('Make sure you have the necessary permissions')
       process.exit(1)
     } else if (/EEXIST/.test(err.message)) {
-      // do nothing
+      // do nothing, but this might be a file, not a directory
+    } else if (/ENOTDIR/.test(err.message)) {
+      console.error('Will not install in a git submodule')
```
Comment (jzaefferer) [inline install.js]: Since we don't consider this an error, should use `console.log` here. Otherwise looks good!

File: lib/sanitize.js [hunk 2]
```diff
@@ -2,20 +2,21 @@ var process = require('child_process')
...
-  if (error.status !== 1) {
-    throw error
-  }
-}
-
 var scissor = /# ------------------------ >8 ------------------------[\s\S]+/
+
 module.exports = function (value) {
+  try {
+    comment = process.execSync(command).toString().trim()
```
Comment (jzaefferer) [inline lib/sanitize.js]: Can you please check/confirm that this runs only once per commitplease process? If we're sanitizing multiple messages, we shouldn't have this call multiple times.

Comment (all3fox) [inline lib/sanitize.js]: You are right, it can get run more than once --- too bad. Will fix that

**Comment by all3fox** [comment]
@jzaefferer you are right that previous `2.7.3-1` was triggering `git config --get core.commentchar` too often (on each message). So there has to be another fix, here is what I tried:

ProjectA that has a `commitplease` as its `devDependency` and ProjectB that has ProjectA as its submodule. When I locally do an `npm install` from inside ProjectA, the `git config --get core.commentchar` returns an `Error` object with `status == 1`, which is suppressed. Looks like when the same command is triggered in the Docker scenario, the status is different, so the error gets rethrown and install fails.

First I thought we could try to figure out the exact status. Then I thought it is too much trouble and will be fragile. Who knows which other status will the same error have in a different environment? So, I decided to suppress all the errors from `git config --get core.commentchar` and just set that char to default `#` in case of any error.

Maybe even remove that call altogether and rely on the fact that "nobody" changes the comment char?

**Comment by jzaefferer** [comment]
Since this causes so much trouble, let's remove. We can add a small note to the Readme, maybe next to the line about git version required for the -v option.

**Comment by all3fox** [comment]
Thanks, I have removed the `core.commentchar` logic altogether. Tell me how to best make this branch look before I merge this to `master`? I was thinking:

 1. Squash f234db6 and 666c8a8 together
 1. Squash 3c7ae3c, 3a8c8a7 and 6d3dee8
 1. Remove all the temporary `2.7.3-x` tags and their respective commits
 1. In a week or two, try to remove all the temporary tags from npm (to give @Jmclerck a heads up and instructions that they have to switch from `beta` or `next` to `2.7.3`) 

So, that should result in three commits: two with code, one with readme changes. Or do you think I just merge this branch as it is?

**Comment by all3fox** [comment]
@jzaefferer I have cleaned up git history (squashing commits and removing temprary tags as described in previous message, no code changes) and ready to merge this to master as `2.7.3`

**Comment by jzaefferer** [comment]
👍 

Could put the last two commits into one - the removal and the README update go together.

**Pull request merged**